src/main/java/capstone/mfslbackend/service/FantasyWeekService.java

```diff
@@ -27,15 +24,9 @@ public FantasyWeek getFantasyWeekById(Long weekId) throws Error404 {
     public List<FantasyWeek> getFantasyWeekByWeekNumber(int weekNumber) {
         return fantasyWeekRepository.findByWeekNumber(weekNumber);
     }
-    public FantasyWeek createFantasyWeek(int fantasyTeamId, int weekNumber) throws Error404, Error400 {
-        if (weekNumber <= 0) {
-            throw new Error400("week number cannot be negative");
-        }
-        FantasyTeam fantasyTeam = fantasyTeamService.getFantasyTeam((long) fantasyTeamId);
-        FantasyWeek fantasyWeek = new FantasyWeek();
-        fantasyWeek.setWeekNumber(weekNumber);
-        fantasyWeek.setFantasyTeam(fantasyTeam);
-        fantasyWeekRepository.save(fantasyWeek);
-        return fantasyWeek;
+
+
+    public FantasyWeek getFantasyWeek(Long weekId) {
+        return fantasyWeekRepository.findById(weekId).orElse(null);
```

Comment:
instead of orElse(null) you should throw a 404 not found error .orElseThrow(() -> new Error404("Could not find fantasy week with id " + weekId))